packages/test/test-service-load/src/runner.ts

```diff
@@ -324,6 +337,78 @@ function scheduleContainerClose(
     });
 }
 
+async function setupOpsMetrics(container: IContainer, logger: ITelemetryLogger, progressIntervalMs: number) {
+    // Use map to cache userName instead of recomputing.
+    const clientIdUserNameMap: { [clientId: string]: string } = {};
+
+    const getUserName = (userContainer: IContainer) => {
+        const clientId = userContainer.clientId;
+        if (clientId !== undefined && clientId.length > 0) {
+            if (clientIdUserNameMap[clientId]) {
+                return clientIdUserNameMap[clientId];
+            }
+
+            const userName: string | undefined = userContainer.getQuorum().getMember(clientId)?.client.user.id;
+            if (userName !== undefined && userName.length > 0) {
+                clientIdUserNameMap[clientId] = userName;
+                return userName;
+            }
+        } else {
+            return "Unknown";
+        }
+    };
+
+    let submitedOps = 0;
+    container.deltaManager.on("submitOp", (message) => {
+        if (message?.type === "op") {
+            submitedOps++;
+        }
+    });
+
+    let receivedOps = 0;
+    container.deltaManager.on("op", (message) => {
+        if (message?.type === "op") {
+            receivedOps++;
+        }
+    });
+
+    let t: NodeJS.Timeout | undefined;
+    const sendMetrics = () => {
+        if (submitedOps > 0) {
+            logger.send({
+                category: "metric",
+                eventName: "Fluid Operations Sent",
+                value: submitedOps,
+                clientId: container.clientId ?? "",
```

Comment:
clientId should never be undefined here - if we ever connected (and submitedOps > 0 tells we connected at least once), then clientId would be always non-undefined after that.

Reply:
Fixed, remove null check.